### PR TITLE
Initial code review.

### PR DESCRIPTION
**Pull request by thepug**

### CODE REVIEW

**Comment by belucid** [comment]
https://github.com/open-company/open-company-search/blob/e792f61603e5f6871e0d9c081597e489f7d31a3d/src/oc/search/elastic_search.clj#L21

Won't we want `with_position_offsets` too for headline? It's the only way we can do hit highlighting since words have been stemmed (and even synonyms potentially) and we need to be able to hit highlight headline and body.

**Comment by belucid** [comment]
@thepug if you delete a board, there's nothing causing the entries to be deleted. Storage service just sends a board delete SNS notification in that case, not an SNS notification for every entry. Search needs to delete every entry with that board ID.

**Comment by belucid** [comment]
@thepug 

We need to discuss search and public/anonymous board access. Presumably this should work, but we need filtering of just public board entries, and the search API is only setup to allow JWT'd access.

Other option (not as good IMHO) is we could punt on search with anonymous/public access for now, but then we need to make sure that's happening in the UI.

**Comment by belucid** [comment]
@thepug 

We're indexing HTML tags from headline and body. You can see this by searching on `br`. Probably need to strip these before indexing through something like:

```
[clj-soup/clojure-soup "0.1.3"]

[jsoup.soup :as soup]

(.text (soup/parse headline))
```
```

**Comment by belucid** [comment]
@thepug right now this is excluding all private entries, even if I do have access to the private board. This is potentially OK for drafts (but even that we'll want to address IMHO), but not really sufficient for private boards. At a minimum we want a Trello card to address the priority for this if we aren't going to do it here.

**Comment by thepug** [comment]
>@thepug

>We need to discuss search and public/anonymous board access. Presumably this should work, but we need filtering of just public board entries, and the search API is only setup to allow JWT'd access.

>Other option (not as good IMHO) is we could punt on search with anonymous/public access for now, but then we need to make sure that's happening in the UI.

I could simply change the query if there is a jwt or not and require an org uuid as a parameter.

**Comment by thepug** [comment]
@belucid  So some html might be worth indexing (media, urls, etc)  Maybe just strip <p> <br> or just allow searching by html?

**Comment by thepug** [comment]
> @thepug right now this is excluding all private entries, even if I do have access to the private board. This is potentially OK for drafts (but even that we'll want to address IMHO), but not really sufficient for private boards. At a minimum we want a Trello card to address the priority for this if we aren't going to do it here.

Maybe I am missing your meaning, but if I create a private board the search result shows items from that board if I have access.  It also excludes if I don't.

**Comment by belucid** [comment]
@thepug 

> Maybe I am missing your meaning, but if I create a private board the search result shows items from that board if I have access. It also excludes if I don't.

In my tests it didn't, but I can retest if this should be working OK.

**Comment by belucid** [comment]
> I could simply change the query if there is a jwt or not and require an org uuid as a parameter.

I don't think it's this simple. Let's discuss in Slack.

**Comment by belucid** [comment]
> @belucid So some html might be worth indexing (media, urls, etc) Maybe just strip or just allow searching by html?

I'll leave this one up to you. Happy with whatever you decide.

**Comment by thepug** [comment]
> In my tests it didn't, but I can retest if this should be working OK.

I found some issues with private boards.  When the private board adds a team member I don't reindex the entries with that change.  I'll fix this.

**Comment by thepug** [comment]
@belucid you are also correct about the filtering for private boards. It isn't working correctly.

**Comment by thepug** [comment]
Trello cards for anonymous user search and fixing the private data filtering.

https://trello.com/c/qNR2iW5A/1141-filter-out-private-entries-in-search-service

https://trello.com/c/mb2QIFxE/1159-allow-anonymous-users-to-search-public-data

File: src/oc/search/elastic_search.clj [hunk 1]
```diff
@@ -238,6 +238,7 @@
                   (add-should-match :slug (:q params)))]
     (timbre/debug "Executing Query:" query)
     (doc/search-all-types conn index {:query query
+                                      :size 20
```
Comment (belucid) [inline src/oc/search/elastic_search.clj]: What does double results returned mean, and why did this solve it?

File: src/oc/search/elastic_search.clj [hunk 2]
```diff
@@ -148,8 +148,28 @@
   [entry-data]
   (add-index "entry" entry-data))
 
+(defn- handle-private-public-board-change
```
Comment (belucid) [inline src/oc/search/elastic_search.clj]: This could benefit from a docstring.

**Comment by belucid** [comment]
Reviewing again.

File: src/oc/search/elastic_search.clj (hunk 1 above)
Comment (thepug) [reply src/oc/search/elastic_search.clj]: I added a docment string in the last commit explaining why only 20 results are returned. The default is 10.  I think we won't need more than 20 and hopefully by the time we do we can add paging support.

**Pull request merged**